crud/common/utils.lua

```diff
@@ -105,19 +106,44 @@ function utils.get_space(space_name, replicasets)
             'perhaps other instances are unavailable or you have configured only the router')
     end
 
-    if replicaset.master == nil then
-        local error_msg = string.format(
-            'The master was not found in replicaset %s, ' ..
-            'check status of the master and repeat the operation later',
-             replicaset.uuid)
-        return nil, GetSpaceError:new(error_msg)
-    end
+    if replicaset.master == nil or replicaset.master.conn.error ~= nil then
+        -- Try to get master again with timeout.
+        local deadline = fiber_clock() + const.GET_SPACE_TIMEOUT
+        local vshard_router, err = utils.get_vshard_router_instance()
```

Comment:
Calling this util without a proper argument is wrong. Either you reuse the router you've got from the outside or extract it the same way it was extracted before. Otherwise it won't work with sharding groups.

Reply:
I propose to consider a current version that passes `vshard_router` inside the `utils.get_space`, after which `vshard_router:routeall()` is called inside timeout loop in `utils.get_space`.

However, there are nuances for `select.lua`, `select_old.lua`, `borders.lua`, `count.lua`:
Before my changes where was one call `local replicasets, err = vshard_router:routeall()`, and `replicasets` passed to `utils.get_space`.

But now there is `vshard_router:routeall()` call inside `utils.get_space` too for these files. 
Perhaps there will be two different versions of `replicasets` values. Now I'm trying to figure out if there is a potential problem here

Reply:
Since replicasets used only to extract util space, I don't see any possible issues (but it would be nice to do a more thorough check, if you wish)